src/editor/overlay_widget.cpp

```diff
@@ -529,6 +529,23 @@ EditorOverlayWidget::fill()
   }
 }
 
+void
+EditorOverlayWidget::replace()
+{
+  auto tilemap = m_editor.get_selected_tilemap();
```

Comment:
This function should utilize `tilemap->save_state()` in the beginning and `tilemap->check_state()` at the end, so the operation can be tracked by undo/redo.